test/units/staking/stakeManager/StakeManager.test.js

```diff
@@ -342,6 +343,85 @@ contract('StakeManager', async function (accounts) {
         testRevert()
       })
     })
+
+    describe('checkSignatures with votes', function () {
+      const amount = new BN(web3.utils.toWei('200'))
+  
+      async function feeCheckpointWithVotes(validatorId, start, end, votes, proposer) {
+        let tree = await buildTreeFee(this.validators, this.accumulatedFees, this.checkpointIndex)
+        this.checkpointIndex++
+  
+        const { data, sigs } = buildSubmitHeaderBlockPaylodWithVotes(
+          this.validatorsWallets[validatorId].getAddressString(),
+          start,
+          end,
+          '' /* root */,
+          Object.values(this.validatorsWallets),
+          votes, // yes votes
+          { rewardsRootHash: tree.getRoot(), allValidators: true, getSigs: true, totalStake: this.totalStaked }
+        )
+  
+        await this.stakeManager.checkSignatures(
+          end - start,
+          utils.bufferToHex(utils.keccak256(Buffer.concat([utils.toBuffer('0x01'), utils.toBuffer(data)]))),
+          utils.bufferToHex(tree.getRoot()),
+          proposer || this.validatorsWallets[validatorId].getAddressString(),
+          sigs,
+          { from: this.rootChainOwner.getAddressString() }
+        )
+  
+        return tree
+      }
+  
+      async function doDeploy() {
+        await freshDeploy.call(this)
+  
+        this.checkpointIndex = 0
+        this.validators = []
+        this.validatorsWallets = {}
+        this.totalStaked = new BN(0)
+        this.accumulatedFees = {}
+  
+        for (let i = 0; i < this.validatorsCount; i++) {
+          await approveAndStake.call(this, { wallet: wallets[i], stakeAmount: amount })
+  
+          const validatorId = i + 1
+          this.validatorsWallets[validatorId] = wallets[i]
+          this.validators.push(validatorId)
+          this.totalStaked = this.totalStaked.add(amount)
+          this.accumulatedFees[validatorId] = []
+        }
+  
+        this.accumSlashedAmount = 0
+        this.index = 0
+      }
+  
+      describe('Deploying and staking with 4 validators...', async function () {
+        const AliceValidatorId = 1
+        const totalFee = new BN(web3.utils.toWei('100'))
+        const firstFeeToClaim = new BN(web3.utils.toWei('25'))
+        const secondFeeToClaim = new BN(web3.utils.toWei('100'))
+  
+        before(function () {
+          this.trees = []
+          this.validatorsCount = 4
+        })
+        before('fresh deploy', doDeploy)
+
+        describe('Alice proposes with more than 2/3+1 votes votes', function () {
+          it('should pass the check', async function () {
```

Comment:
add 2 more tests for 2/3+1 votes and sigPrefix[0x00, 0x02]

Reply:
done